src/main/java/org/gitlab4j/api/GitLabApiClient.java

```diff
@@ -600,23 +600,37 @@ protected Response upload(String name, File fileToUpload, String mediaTypeString
      * @throws IOException if an error occurs while constructing the URL
      */
     protected Response upload(String name, File fileToUpload, String mediaTypeString, Form formData, URL url) throws IOException {
+        MediaType mediaType = (mediaTypeString != null ? MediaType.valueOf(mediaTypeString) : null);
+        FileDataBodyPart filePart = mediaType != null ?
```

Comment:
After checking in `FileDataBodyPart` constructors, you don't need to test if mediaType is null or not. In fact the constructor used line 606 call the one used line 605 with null in MediaType. Please remove the ternary operator.

Reply:
Oh yes, I see.
Nice, we can rely on `FileDataBodyPart`'s capability of detecting the media type from a `File`.

The method parameter `mediaTypeString` should be removed then.
Since it's declared as `protected` it is exposed to clients, so we may not remove it until we get to a major version, isn't it?
I'll just update the javadocs for now.